code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,308 @@
+/obj/machinery/deployable/mounted/proc/start_fire(datum/source, atom/object, turf/location, control, params)
+	SIGNAL_HANDLER
+
+	update_icon_state()
+
+	if(gun.gun_on_cooldown(operator))
+		return
+
+	if(QDELETED(object))
+		return
+
+	if(object == src) //Clicking the gun gets off of it.
+		var/mob/living/carbon/human/user = source
+		user.unset_interaction()
+		return
+
+	var/target = object
+
+	var/list/modifiers = params2list(params)
+	if(istype(target, /obj/screen))
+		if(!istype(target, /obj/screen/click_catcher))
+			return
+		//Happens when you click a black tile
+		target = params2turf(modifiers["screen-loc"], get_turf(operator), operator.client)
+
+	if(!can_fire(target))
+		return
+
+	gun.set_target(target)
+
+	if(modifiers["right"] || modifiers["middle"] || modifiers["shift"])
+		if(gun.active_attachable?.flags_attach_features & ATTACH_WEAPON)
+			gun.do_fire_attachment()
+		return
+	if(gun.gun_firemode == GUN_FIREMODE_SEMIAUTO)
+		if(!gun.Fire(TRUE) || gun.windup_checked == WEAPON_WINDUP_CHECKING)
+			return
+		gun.reset_fire()
+		return
+	gun.gun_user.client.mouse_pointer_icon = 'icons/effects/supplypod_target.dmi'
+	SEND_SIGNAL(gun, COMSIG_GUN_FIRE)
+
+///Happens when you drag the mouse.
+obj/machinery/deployable/mounted/proc/change_target(datum/source, atom/src_object, atom/over_object, turf/src_location, turf/over_location, src_control, over_control, params)
+	SIGNAL_HANDLER
+	if(istype(over_object, /obj/screen))
+		if(!istype(over_object, /obj/screen/click_catcher))
+			return
+		var/list/modifiers = params2list(params)
+		over_object = params2turf(modifiers["screen-loc"], get_turf(operator), operator.client)
+
+	if(!can_fire(over_object))
+		return
+
+	gun.set_target(over_object)
+	operator.face_atom(over_object)
+
+///Checks if you can fire
+/obj/machinery/deployable/mounted/proc/can_fire(atom/object)
+
+	if(operator.lying_angle || !Adjacent(operator) || operator.incapacitated() || get_step(src, REVERSE_DIR(dir)) != operator.loc)
+		operator.unset_interaction()
+		return FALSE
+	if(operator.get_active_held_item())
+		to_chat(operator, "<span class='warning'>You need a free hand to shoot the [src].</span>")
+		return FALSE
+
+	var/atom/target = object
+
+	if(!istype(target))
+		return FALSE
+
+	if(isnull(operator.loc) || isnull(loc) || !z || !target?.z == z)
+		return FALSE
+
+
+	var/angle = get_dir(src, target)
+	var/direction = dir
+	//we can only fire in a 90 degree cone
+	if((direction & angle) && target.loc != loc && target.loc != operator.loc)
+		operator.setDir(direction)
+		gun.set_target(target)
+		return TRUE
+	else if (!(direction & angle) && target.loc != loc && target.loc != operator.loc) // Let us rotate this stuff.
+		if(deploy_flags & DEPLOYED_NO_ROTATE)
+			to_chat(operator, "This one is anchored in place and cannot be rotated.")
+			return FALSE
+
+		var/list/leftright = LeftAndRightOfDir(direction)
+		var/left = leftright[1] - 1
+		var/right = leftright[2] + 1
+		if(left == (angle-1) || right == (angle+1))
```

Comment:
this code is cryptic, but i guess it's old code. Don't think you need to change it